### PR TITLE
Feature/s 5 schedule as calendar

### PR DESCRIPTION
**Pull request by keksik77**
Implement calendar component

image
image

### CODE REVIEW

File: client/src/components/Schedule/CalendarView.tsx [hunk 1]
```diff
@@ -5,7 +5,9 @@ type Props = {
   timeZone: string;
 };
 
+
 export function CalendarView({ data, timeZone }: Props) {
+  console.log(data);
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView.tsx]: Don't forget to remove log before merging

File: client/src/components/Schedule/CalendarView/Calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import DesktopCalendar from './components/DesktopCalendar';
+import { CourseEvent } from 'services/course';
+
+type Props = {
+  data: CourseEvent[];
+  timeZone: string;
+};
+
+export const CalendarView: React.FC<Props> = ({data, timeZone}) => {
+  const { width } = useWindowDimensions();
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/Calendar.tsx]: If I'm not mistaken we already use `mobile-device-detect` package. Let's keep on using it through entire project. And there is no need to listen resize event. We need to detect the size only on first load.

File: client/src/components/Schedule/CalendarView/components/DesktopCalendar.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          )})}
+        </ul>
+      </Scrollbars>
+    );
+  }
+
+  const monthCellRender = (date: unknown | Moment) => {
+    const num = getMonthValue(date as unknown as Moment, data, timeZone);
+
+    return num ? (
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/components/DesktopCalendar.tsx]: Can be simplified: `!!num && (...)`

File: client/src/components/Schedule/CalendarView/components/MobileCalendar.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+              position: absolute;
+              bottom: 12px;
+              background: #FB6216;
+              right: -13px;
+              border-radius: 100%;
+              width: 20px;
+              height: 20px;
+              line-height: 20px;
+              color: white;
+            }
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/components/MobileCalendar.tsx]: Styles for month cell and data cell look pretty similar. let's move it to its own constant.

File: client/src/components/Schedule/CalendarView/components/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  isOpen: boolean;
+  dataEvent: CourseEvent;
+  handleOnClose: Function;
+  timeZone: string;
+};
+
+export function ModalWindow({ isOpen, dataEvent, handleOnClose, timeZone }: Props) {
+
+  //!plug
+  const newTimeZone = timeZone === 'Europe/Yekaterinburg' ? 'Asia/Yekaterinburg' : timeZone;
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/components/Modal.tsx]: Let's fix it directly in `client/src/configs/timezones.ts`

File: client/src/components/Schedule/CalendarView/utils/DataFuncs.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { filterByDate } from './DateFuncs';
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/utils/DataFuncs.ts]: Let's rename the file to `filters.ts` and join it with the file below.

File: client/src/components/Schedule/CalendarView/utils/DataFuncs.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import { CourseEvent } from 'services/course';
+import { EventTypeColor} from 'components/Schedule/model';
+import {Moment} from 'moment';
+
+
+export function getListData(calendarCellDate: Moment, data: CourseEvent[], timeZone: string) {
+  const listData: { color: string; name: string; key: number }[] = [];
+
+  filterByDate(calendarCellDate, data, timeZone).forEach((el: CourseEvent) => {
+    const tagColor = EventTypeColor[el.event.type as keyof typeof EventTypeColor];
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/utils/DataFuncs.ts]: WDYT about rewriting this function in more functional way? 

```
return filterByDate(calendarCellDate, data, timeZone).map((event: CourseEvent) => {
  const tagColor = EventTypeColor[el.event.type as keyof typeof EventTypeColor];
  return { color: tagColor || '#d9d9d9' , name: el.event.name, key: el.id };
});
```

Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/utils/DataFuncs.ts]: I'm sure, that the construction `[el.event.type as keyof typeof EventTypeColor]` can be rewritten to something far simpler.

File: client/src/components/Schedule/CalendarView/utils/DateFuncs.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  const newTimeZone = timeZone === 'Europe/Yekaterinburg' ? 'Asia/Yekaterinburg' : timeZone;
+
+  return data.filter(
+    (event) => calendarCellDate.format('YYYY-MM-DD') === moment(event.dateTime, 'YYYY-MM-DD HH:mmZ').tz(newTimeZone).format('YYYY-MM-DD'),
+  );
+}
+
+export function getMonthValue(calendarCellDate: Moment, data: CourseEvent[], timeZone: string) {
+  //plug!
+  const newTimeZone = timeZone === 'Europe/Yekaterinburg' ? 'Asia/Yekaterinburg' : timeZone;
```
Comment (yuliaHope) [inline client/src/components/Schedule/CalendarView/utils/DateFuncs.ts]: Will be redundant after moving the fix for time zone to the root.

File: client/src/components/Schedule/CalendarView/Calendar.tsx (hunk 2 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/Calendar.tsx:13]: If you reduce the browser window, the calendar will not a usable, are you sure?
image

File: client/src/components/Schedule/CalendarView.tsx (hunk 1 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView.tsx]: Done

File: client/src/components/Schedule/CalendarView/Calendar.tsx (hunk 2 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/Calendar.tsx]: Done

File: client/src/components/Schedule/CalendarView/components/DesktopCalendar.tsx (hunk 3 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/components/DesktopCalendar.tsx]: Done

File: client/src/components/Schedule/CalendarView/components/MobileCalendar.tsx (hunk 4 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/components/MobileCalendar.tsx]: Done

File: client/src/components/Schedule/CalendarView/components/Modal.tsx (hunk 5 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/components/Modal.tsx]: Done

File: client/src/components/Schedule/CalendarView/utils/DataFuncs.ts (hunk 6 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/utils/DataFuncs.ts]: Done

File: client/src/components/Schedule/CalendarView/utils/DataFuncs.ts (hunk 7 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/utils/DataFuncs.ts]: Done

File: client/src/components/Schedule/CalendarView/utils/DateFuncs.ts (hunk 8 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/utils/DateFuncs.ts]: Done

File: client/src/components/Schedule/CalendarView/utils/DataFuncs.ts (hunk 7 above)
Comment (keksik77) [reply client/src/components/Schedule/CalendarView/utils/DataFuncs.ts]: I didn't succeed(

**Review by yuliaHope — APPROVED** [review]

**Pull request merged**